tests/scenario_tests/test_error_handler.py

```diff
@@ -153,3 +154,91 @@ def failing_listener():
         response = app.dispatch(request)
         assert response.status == 500
         assert response.headers["x-test-result"] == ["1"]
+
+    def test_unhandled_errors(self):
+        app = App(
+            client=self.web_client,
+            signing_secret=self.signing_secret,
+            raise_error_for_unhandled_request=True,
+        )
+        response = app.dispatch(self.build_valid_request())
+        assert response.status == 404
+        assert response.body == '{"error": "unhandled request"}'
+
+        @app.error
+        def handle_errors(error):
+            assert isinstance(error, BoltUnhandledRequestError)
+            return BoltResponse(status=404, body="TODO")
```

Comment:
Just curious, about how this test would fail. Let's pretend that there was a bug causing `raise_error_for_unhandled_request=True` to not raise the error. In that case, `handle_errors` would not be triggered and the assertion inside would not run.

In other test frameworks, we can "expect 4 asserts" to guarantee to catch a situation where callback pay not be triggered.

How does it work with Python in this case? I'm concerned that the assertion on L170 may not trigger but the test will pass.

Reply:
@mwbrooks 
>In that case, handle_errors would not be triggered and the assertion inside would not run.

L175 verifies if the response body is "TODO". As the default body is '{"error": "unhandled request"}', the assertion fails if this listener is not executed.

>In other test frameworks, we can "expect 4 asserts" to guarantee to catch a situation where callback pay not be triggered.

If the assertion in the `handle_errors` method fails, it raises an exception. As a result, the Bolt app responds with status 500. 

Regarding the other assertions in the test method, there is nothin special. If any of them fails, the test suite fails too.